v0/destinations/tiktok_ads/transform.js

```diff
@@ -52,6 +53,20 @@ const trackResponseBuilder = async (message, { Config }) => {
   if (isDefinedAndNotNullAndNotEmpty(externalId)) {
     set(payload, "context.user.external_id", externalId);
   }
+
+  const traits = getFieldValueFromMessage(message, "traits");
+
+  // taking user properties like email and phone from traits
+  let email = get(payload, "context.user.email");
+  if (!isDefinedAndNotNullAndNotEmpty(email) && traits?.email) {
+    set(payload, "context.user.email", traits.email);
+  }
+
+  let phone_number = get(payload, "context.user.phone_number");
+  if (!isDefinedAndNotNullAndNotEmpty(phone_number) && traits?.phone) {
+    set(payload, "context.user.phone_number", traits.phone);
+  }
+
```

Comment:
It is not the right way to collect user traits. We get user traits in context.traits after an identify call. We should map that to context.user in the resultant payload, not from properties.context.user.

Reply:
We are taking traits from [`traits`, `context.traits`] L:57. and mapping that to `context.user.xxx` in resultant payload

Reply:
We just have email and phone. Ref - https://ads.tiktok.com/gateway/docs/index?identify_key=2b9b4278e47b275f36e7c39a4af4ba067d088e031d5f5fe45d381559ac89ba48&language=ENGLISH&doc_id=1727541103358977#item-link-Context%20object%20parameters

Reply:
Accha got it.